### PR TITLE
feat: components

### PR DESCRIPTION
**Pull request by pi0**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

👀

**Core fix:** Don't use module handler fn name as unique key so we can do unique module check by default. This is important as one may add components module to modules/buildModules section

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Review by kevinmarrec — DISMISSED** [review]
😍

File: packages/config/src/options.js [hunk 1]
```diff
@@ -448,6 +448,11 @@ export function getNuxtConfig (_options) {
       .map(([path, handler]) => ({ path, handler }))
   }
 
+  // Components module
+  if (options.components) {
+    options.buildModules.push('@nuxt/components')
```
Comment (pi0) [inline packages/config/src/options.js:453]: Side note: This is safe if user already specify in `modules` or `buildModules` section ([src](https://github.com/nuxt/nuxt.js/blob/dev/packages/core/src/module.js#L187)).

**Review by atinux — APPROVED** [review]
Please add `[release]` when merging :)

**Comment by Atinux** [comment]
Added the reference in upcoming docs: https://github.com/nuxt/docs/pull/1922/commits/46c6ba8a29ecbc1553d8adf08e8d1a96a796ffab

**Pull request merged**